posthog/models/insight.py

```diff
@@ -154,6 +154,13 @@ def dashboard_filters(self, dashboard: Optional[Dashboard] = None):
         else:
             return self.filters
 
+    def dashboard_query(self, dashboard: Optional[Dashboard]) -> dict:
+        if not dashboard or not self.query:
+            return self.query
```

Comment:
This is potentially breaking the return type. It's basically `if not self.query: return self.query`. However self.query could also be `None`, in which case the `-> dict` return type is wrong.

Reply:
Ah I'm sort of surprised Python let me do this! Changed to `-> Optional[dict]`

Reply:
Yeah, python's typing is dodgy at best. However here's a hot tip: explicitly write `self.query is not None` if you want to check for nulls.